src/components/RefreshButton.tsx

```diff
@@ -1,26 +1,17 @@
 import { Button } from '@chakra-ui/react';
 
-import { DispatchWithAutoAbortProps } from '@/hooks/useDispatchWithAutoAbort';
-import useHandleRefresh, { ActionBuilderT } from '@/hooks/useHandleRefresh';
-
 const RefreshButton = ({
     loading,
-    actionBuilder,
-    dispatchWithAutoAbort,
+    list,
 }: {
     loading: boolean;
-    actionBuilder: ActionBuilderT;
-    dispatchWithAutoAbort: DispatchWithAutoAbortProps;
+    list: () => void;
```

Comment:
Given all the changes here, I'd rewrite the component as a simple:
```
import { Button, ButtonProps } from '@chakra-ui/react';

const RefreshButton = (props: ButtonProps): JSX.Element => {
    return (
        <Button
            size="sm"
            variant="outline"
            loadingText="Loading"
            {...props}
        >
            Refresh
        </Button>
    );
};

export default RefreshButton;
```
No need to define specific list and loading props, isLoading and onClick do just as well!

Reply:
👍  list is a remnant of when i was handling both list from zustand and redux while making the change